### PR TITLE
Improve documentation

### LINKED ISSUE #65: Missing docstrings
Most of the sources are lacking docstrings.
It would be useful to add short comments for each function/struct.

### PR DESCRIPTION
**Pull request by sorru94**
Adds docstrings to most public and private functions.
Closes #65

### CODE REVIEW

File: proto/astarteplatform/msghub/astarte_message.proto [hunk 1]
```diff
@@ -26,14 +26,16 @@ import "google/protobuf/timestamp.proto";
...
-  string interface_name = 1;
-  string path = 2;
-  oneof payload {
-    AstarteDataType astarte_data = 3;
-    AstarteUnset astarte_unset = 4;
+  string interface_name = 1; // Name of the interface to send data on.
+  string path = 2;           // Endpoint to send the data on.
+  oneof payload {            // Content of the message.
+    AstarteDataType astarte_data = 3; // Effective payload.
+    AstarteUnset astarte_unset = 4;   // Null payload.
```
Comment (harlem88) [inline proto/astarteplatform/msghub/astarte_message.proto:35]: Choose whether to leave a single space after the semicolon or to line it up with the other comments.

File: proto/astarteplatform/msghub/node.proto [hunk 2]
```diff
@@ -22,7 +22,8 @@ syntax = "proto3";
 
 package astarteplatform.msghub;
 
+/* This message defines a node to be attached/detached to the Astarte message hub. */
 message Node {
-  string uuid = 1;
-  repeated bytes interface_jsons = 2;
+  string uuid = 1;                    // The node identifier.
+  repeated bytes interface_jsons = 2; // Serialized .json interface file as bytes array.
```
Comment (harlem88) [inline proto/astarteplatform/msghub/node.proto:28]: same above.

File: src/data/astarte.rs [hunk 3]
```diff
@@ -17,6 +17,13 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
+//! The message hub uses an independent handler to communicate with Astarte.
```
Comment (harlem88) [inline src/data/astarte.rs:20]:
```suggestion
//! The Astarte message hub uses an independent handler to communicate with Astarte.
```

File: src/data/astarte_provider.rs [hunk 4]
```diff
@@ -17,33 +17,38 @@
...
+use astarte_sdk::Interface;
+
 use crate::astarte_message_hub::AstarteNode;
 use crate::data::astarte::AstarteSubscriber;
 #[cfg(test)]
 use crate::data::mock_astarte::MockAstarteSdk as AstarteSdk;
 use crate::error::AstarteMessageHubError;
 use crate::proto_message_hub::AstarteMessage;
 
+/// An Astarte SDK based implementation of an Astarte handler.
```
Comment (harlem88) [inline src/data/astarte_provider.rs:44]:
```suggestion
/// An Astarte Device SDK based implementation of an Astarte handler.
```

File: src/data/astarte_provider.rs [hunk 5]
```diff
@@ -17,33 +17,38 @@
...
+/// Uses the Astarte SDK to provide subscribe and publish functionality.
```
Comment (harlem88) [inline src/data/astarte_provider.rs:45]:
```suggestion
/// Uses the Astarte Device SDK to provide subscribe and publish functionality.
```

File: src/data/astarte_provider.rs [hunk 6]
```diff
@@ -84,6 +89,15 @@ impl AstarteSubscriber for Astarte {
 }
 
 impl Astarte {
+    /// Runner function for the Astarte handler.
+    ///
+    /// Polls the Astarte SDK for received messages. When the received message interface
```
Comment (harlem88) [inline src/data/astarte_provider.rs:94]:
```suggestion
    /// Polls the Astarte Device SDK for received messages. When the received message interface
```

File: src/data/mock_astarte.rs [hunk 7]
```diff
@@ -17,10 +17,12 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
+//! Contains a mock version of the Astarte SDK to be used for unit testing.
```
Comment (harlem88) [inline src/data/mock_astarte.rs:20]:
```suggestion
//! Contains a mock version of the Astarte Device SDK to be used for unit testing.
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: proto/astarteplatform/msghub/node.proto (hunk 2 above)
Comment (FrancescoVaiani) [inline proto/astarteplatform/msghub/node.proto:28]:
```suggestion
  repeated bytes interface_jsons = 2; // Array of byte arrays representing all .json interface files of the node.
```

**Review by FrancescoVaiani — COMMENTED** [review]
A little note about something I myself found confusing

**Comment by sorru94** [comment]
Succeeded by #111.

**Pull request closed without merging**